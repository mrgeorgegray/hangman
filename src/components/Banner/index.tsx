/** @jsxImportSource @emotion/react */
import React from "react";
import { css } from "@emotion/react";

interface BannerProps {
  action: JSX.Element;
  message: string;
  type: "error" | "success";
}

const Banner: React.FC<BannerProps> = ({ action, message, type }) => (
  <div
    css={css`
      border: 2px solid #30353b;
      padding: 20px;
      margin-top: 20px;
      max-width: 1000px;
      text-align: center;
      ${type === "error" && ` background-color: #FFBABA; `}
      ${type === "success" && ` background-color: #DFF2BF; `}

      @media (min-width: 720px) {
        left: 0;
        margin-left: auto;
        margin-right: auto;
        margin-top: 0px;
        position: absolute;
        right: 0;
        top: 10px;
        width: 90%;
      }
    `}
  >
    <span
      css={css`
        display: block;
        margin-bottom: 20px;

        @media (min-width: 720px) {
          display: inline;
          margin-right: 20px;
          margin-bottom: 0px;
        }
      `}
    >
      {message}
    </span>
    {action}
  </div>
);

export default Banner;
