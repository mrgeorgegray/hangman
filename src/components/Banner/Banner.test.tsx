import React from "react";
import { render } from "@testing-library/react";

import Banner from ".";

describe("<Banner />", () => {
  it("renders a error state", () => {
    const { container } = render(<Banner type="error" text="uh oh" />);
    expect(container).toMatchSnapshot();
  });

  it("renders a success state", () => {
    const { container } = render(<Banner type="success" text="woohoo" />);
    expect(container).toMatchSnapshot();
  });
});
