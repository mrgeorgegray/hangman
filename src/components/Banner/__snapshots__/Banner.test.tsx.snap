// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders a error state 1`] = `
<div>
  <div
    class="css-1vttx9d-Banner"
  >
    <span
      class="css-1w6p0kp-Banner"
    />
  </div>
</div>
`;

exports[`<Banner /> renders a success state 1`] = `
<div>
  <div
    class="css-icnjft-Banner"
  >
    <span
      class="css-1w6p0kp-Banner"
    />
  </div>
</div>
`;
